lib/Dialect/FIRRTL/Transforms/LowerLayers.cpp

```diff
@@ -384,10 +384,12 @@ LowerLayersPass::runOnModuleLike(FModuleLike moduleLike) {
 
 void LowerLayersPass::lowerInlineLayerBlock(LayerOp layer,
                                             LayerBlockOp layerBlock) {
-  OpBuilder builder(layerBlock);
-  auto macroName = macroNames[layer];
-  auto ifDef = sv::IfDefOp::create(builder, layerBlock.getLoc(), macroName);
-  ifDef.getBodyRegion().takeBody(layerBlock.getBodyRegion());
+  if (!layerBlock.getBody()->empty()) {
+    OpBuilder builder(layerBlock);
+    auto macroName = macroNames[layer];
+    auto ifDef = builder.create<sv::IfDefOp>(layerBlock.getLoc(), macroName);
+    ifDef.getBodyRegion().takeBody(layerBlock.getBodyRegion());
+  }
```

Comment:
Not creating the empty ifdefs unnecessarily seems good.

Out of curiosity, do we canonicalize these away in SV if they are empty today?

Reply:
Yep there is a canonicalizer.